src/libasr/asr_verify.cpp

```diff
@@ -303,6 +303,10 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
         }
     }
 
+    void visit_UnionType(const UnionType_t& x) {
+        visit_DerivedTypeEnumTypeUnionType(x);
+    }
```

Comment:
Are there any requirements on Union besides the struct?

Reply:
I don't think. On the syntax and ASR level, DerivedType and Union are exactly the same. Its in the backend level where the difference comes into play, DerivedType stores all the members in memory but Union stores only the maximum sized element.